collector/lib/ProcfsScraper.cpp

```diff
@@ -114,9 +127,12 @@ bool GetSocketINodes(int dirfd, UnorderedSet<ino_t>* sock_inodes) {
   while (auto curr = fd_dir.read()) {
     if (!std::isdigit(curr->d_name[0])) continue;  // only look at fd entries, ignore '.' and '..'.
 
-    ino_t inode;
-    if (!ReadINode(fd_dir.fd(), curr->d_name, "socket", &inode)) continue;  // ignore non-socket fds
-    sock_inodes->insert(inode);
+    SocketInfo sock_info;
+    sock_info.pid = pid;
+
+    if (!ReadINode(fd_dir.fd(), curr->d_name, "socket", &(sock_info.inode))) continue;  // ignore non-socket fds
```

Comment:
[nit] Think we could do this check before constructing `sock_info` if we store `sock_info.inode` on a temp variable, then we can directly construct `sock_info` and insert it into sock_inodes.

Reply:
I replaced this sequence by a call to emplace() which prevents any work prior to the socket test.